src/inodefilter.rs

```diff
@@ -0,0 +1,76 @@
+
+#![cfg_attr(windows, feature(windows_by_handle))]
+
+use std::collections::HashMap;
+
+#[derive(Debug, Default, Clone)]
+pub struct InodeFilter {
+    inner: HashMap<u64, u64>
```

Comment:
Just out of curiousity: Isn't a BTreeMap faster in this case? The way I understand it: HashMaps first hash their input, and then stuff that into a tree to do comparisons. u64 naturally has that ability, so the Hash part seems superfluous.
What am I missing?

Reply:
It seems that a BTreeMap is actually a tad slower, but could be within the margin of error. My dataset took 35s to iterate with default options, and the BTreeMap came in ~400ms slower.

Reply:
I suspected a tree might be slower due to either more pointer chasing or the cost of its linear searches in buckets vs the hashmap's direct indexing, but wasn't sure either way, and it could easily vary across systems depending on the relative performance of CPU/memory.

The HashMap was the more conservative option, since the additional features of a tree weren't needed, and I doubt the difference matters much at this scale.

Reply:
Thanks for sharing! As always, much appreciated, I learned something.